pg_shard.c

```diff
@@ -537,9 +539,17 @@ ErrorIfQueryNotSupported(Query *queryTree)
 				specifiesPartitionValue = true;
 			}
 		}
+
+		whereClause = ((queryTree->jointree == NULL) ?
+					   NULL : queryTree->jointree->quals);
+		if (contain_mutable_functions(whereClause) ||
+			contain_volatile_functions(whereClause))
```

Comment:
I think it's sufficient to call `contain_mutable_function` here: anything that is `VOLATILE` is also not `IMMUTABLE`, so this is redundant.